packages/ui-components/src/components/Map/InteractivePolygon.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Polygon } from 'react-leaflet';
+import styled from 'styled-components';
+import { AreaTooltip } from './AreaTooltip';
+import { MAP_COLORS, BREWER_PALETTE } from './constants';
+import ActivePolygon from './ActivePolygon';
+
+const { POLYGON_BLUE, POLYGON_HIGHLIGHT } = MAP_COLORS;
+
+const BasicPolygon = styled(Polygon)`
+  fill: ${POLYGON_BLUE};
+  fill-opacity: 0.04;
+  stroke-width: 1;
+  stroke: ${POLYGON_BLUE};
+
+  :hover {
+    fill-opacity: 0.5;
+    stroke: ${POLYGON_HIGHLIGHT};
+    fill: ${POLYGON_HIGHLIGHT};
+  }
+`;
+
+export const ShadedPolygon = styled(Polygon)`
+  fill-opacity: 0.5;
+
+  &:hover {
+    fill-opacity: 0.8;
+  }
+`;
+
+const parseProps = (organisationUnitCode, organisationUnitChildren, measureOrgUnits) => {
+  let shade;
+  let isHidden;
+  let orgUnitMeasureData;
+  let hasShadedChildren = false;
+
+  if (measureOrgUnits.length > 0) {
+    const measureOrgUnitCodes = measureOrgUnits.map(orgUnit => orgUnit.organisationUnitCode);
```

Comment:
A simple change to make this code block more efficient would be to use a Set here and a `.has` hashed lookup wherever `.includes` is currently running a full loop across the array
```suggestion
    const measureOrgUnitCodes = new Set(measureOrgUnits.map(orgUnit => orgUnit.organisationUnitCode));
```

Reply:
Great suggestion. I will start using Sets more